### PR TITLE
Install NVIDIA pre-built modules for CachyOS kernels instead of nvidia-dkms

### PR DESCRIPTION
**Pull request by ventureoo**
Depends on: https://github.com/CachyOS/chwd/pull/56

### CODE REVIEW

**Comment by vnepogodin** [comment]
@ventureoo what should we do in case system have multiple kernels installed, including kernels which dont provide `-nvidia` package or which are arch kernels( which provide `-nvidia`)?

**Comment by ventureoo** [comment]
> @ventureoo what should we do in case system have multiple kernels installed, including kernels which dont provide `-nvidia` package or which are arch kernels( which provide `-nvidia`)?

Well, I thought about it. Originally, I wanted to install nvidia-dkms along with the pre-built modules for our kernels, but the problem is that modules built by nvidia-dkms probably overrides pre-compiled modules, making it pointless to install them. So I decided to install nvidia-dkms only if there are no available cachyos kernels on the system. I can also add a check that enables `nvidia` module for the arch kernel if it is installed. But other kernels that don't do this are probably screwed.

On the other hand, should we even be worried about it? The chwd script is most often used to install drivers during the system installation process, and we do not offer kernels other than our own for installation.  If a user has installed CachyOS, they should use the kernels we offer, not someone else's.

**Comment by vnepogodin** [comment]
> On the other hand, should we even be worried about it? The chwd script is most often used to install drivers during the system installation process, and we do not offer kernels other than our own for installation. If a user has installed CachyOS, they should use the kernels we offer, not someone else's.

yeah I agree with that.

> Well, I thought about it. Originally, I wanted to install nvidia-dkms along with the pre-built modules for our kernels, but the problem is that modules built by nvidia-dkms probably overrides pre-compiled modules, making it pointless to install them. So I decided to install nvidia-dkms only if there are no available cachyos kernels on the system. I can also add a check that enables nvidia module for the arch kernel if it is installed. But other kernels that don't do this are probably screwed.

well it means that CachyOS kernels which dont provide -nvidia pre-compiled module, the installation would be corrupt due to not having kernel module resulting in "target not found" during installatiion

**Comment by ventureoo** [comment]
> > On the other hand, should we even be worried about it? The chwd script is most often used to install drivers during the system installation process, and we do not offer kernels other than our own for installation. If a user has installed CachyOS, they should use the kernels we offer, not someone else's.
> 
> yeah I agree with that.
> 
> > Well, I thought about it. Originally, I wanted to install nvidia-dkms along with the pre-built modules for our kernels, but the problem is that modules built by nvidia-dkms probably overrides pre-compiled modules, making it pointless to install them. So I decided to install nvidia-dkms only if there are no available cachyos kernels on the system. I can also add a check that enables nvidia module for the arch kernel if it is installed. But other kernels that don't do this are probably screwed.
> 
> well it means that CachyOS kernels which dont provide -nvidia pre-compiled module, the installation would be corrupt due to not having kernel module resulting in "target not found" during installatiion

I don't think this is a problem, as all kernels currently have NVIDIA pre-built module (since it is part of kernel PKGBUILD).

File: pci/graphic_drivers/profiles.toml [hunk 1]
```diff
@@ -27,7 +27,7 @@ conditional_packages = """
     if [ -n "$kernels" ]; then
         for kernel in $kernels; do
             case "$kernel" in
-                *-headers|*-nvidia|*-zfs);;
+                *-headers|*-zfs);;
```
Comment (vnepogodin) [inline pci/graphic_drivers/profiles.toml]: this results in `linux-cachyos-nvidia linux-cachyos-nvidia-nvidia`
if nvidia module already installed

Comment (ventureoo) [reply pci/graphic_drivers/profiles.toml]: yep, my bad. We still need to add packages to the list that are installed when we remove them. https://github.com/CachyOS/chwd-db/pull/26/commits/57c78cb241f6538e3602296cabfd0b7eaec4c445 should fix that.

**Comment by ptr1337** [comment]
Personally im fine with this, but we should consider to create PKGBUILDs for linux, linux-zen and maybe linux-lts (or just a -nvidia PKGBUILD, which builds it with fakeroot and dkms).
Second method could be problematic, if arch is updating the kernel and we update the nvidia module with a delay.

Adding these packages only to the "cachyos" (v1) repository, should fix the above mentioned issues which vlad has mentioned.

WDYT?
@vnepogodin @ventureoo

**Comment by ventureoo** [comment]
I add module installation for stock arch kernels as well.

**Comment by vnepogodin** [comment]
> Personally im fine with this, but we should consider to create PKGBUILDs for linux, linux-zen and maybe linux-lts (or just a -nvidia PKGBUILD, which builds it with fakeroot and dkms). Second method could be problematic, if arch is updating the kernel and we update the nvidia module with a delay.
> 
> Adding these packages only to the "cachyos" (v1) repository, should fix the above mentioned issues which vlad has mentioned.
> 
> WDYT? @vnepogodin @ventureoo

arch already provide pre-built nvidia module for `linux`, `linux-lts`.

I don't think its necessary to provide separate PKGBUILDs for them per se

**Comment by ptr1337** [comment]
> > Personally im fine with this, but we should consider to create PKGBUILDs for linux, linux-zen and maybe linux-lts (or just a -nvidia PKGBUILD, which builds it with fakeroot and dkms). Second method could be problematic, if arch is updating the kernel and we update the nvidia module with a delay.
> > Adding these packages only to the "cachyos" (v1) repository, should fix the above mentioned issues which vlad has mentioned.
> > WDYT? @vnepogodin @ventureoo
> 
> arch already provide pre-built nvidia module for `linux`, `linux-lts`.
> 
> I don't think its necessary to provide separate PKGBUILDs for them per se

Yeah, probably it is fine then. Actually since chwd does only run at the first installation process, we don't need to add this.
The only reason to add a own "linux-zen" would be to provide for this kernel a nvidia module.

From my side this is fine, maybe we need to check the ISO and calamares, if there are any changes required.
On the ISO we need to switch from nvidia-dkms to linux-cachyos-nvidia

**Comment by vnepogodin** [comment]
> Yeah, probably it is fine then. Actually since chwd does only run at the first installation process, we don't need to add this. The only reason to add a own "linux-zen" would be to provide for this kernel a nvidia module.

Yep. I answered [here](https://github.com/CachyOS/chwd-db/pull/26#issuecomment-1807104043) about it.
that we are fine with handling only CachyOS kernels.

**Comment by vnepogodin** [comment]
> From my side this is fine, maybe we need to check the ISO and calamares, if there are any changes required.
> On the ISO we need to switch from nvidia-dkms to linux-cachyos-nvidia

yeah only small changes required

**Comment by vnepogodin** [comment]
> I add module installation for stock arch kernels as well.

LGTM

**Pull request merged**